### PR TITLE
[v11.3.x] Alerting Docs: Update Alert rule and Notification Templating

### PR DESCRIPTION
**Pull request by ppcano**
Backport 813632dff13f69c84daf328a6562c660d1b19016 from #94069

---

This PR is an entire restructurating of the Alerting Templating documentation. It continues the work started in https://github.com/grafana/grafana/pull/93935, which updated the Introduction to Templates documentation.

https://github.com/grafana/grafana/pull/94711 relates to this PR, adding the same notification template examples in the Grafana UI.

Screenshot 2024-10-01 at 16 29 54

Closes https://github.com/grafana/grafana/issues/93084
Closes https://github.com/grafana/support-escalations/issues/12383
Closes https://github.com/grafana/support-escalations/issues/6315

### CODE REVIEW

**Review by tonypowa — APPROVED** [review]

File: docs/sources/alerting/alerting-rules/templates/language.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+At the start of notification templates, dot (`.`) refers to [Notification Data](ref:reference-notificationdata).
+
+```go
+{{ .Alerts }}
+```
+
+In annotation and label templates, dot (`.`) is initialized with all alert data. It’s recommended to use the [`$labels` and `$values` variables](ref:alert-rule-template-reference-variables) instead to directly access the alert labels and query values.
+
+{{% admonition type="note" %}}
+Dot (`.`) might refer to something else when used in a [range](#range), a [with](#with), or when writing [templates](#templates) used in other templates.
```
[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/alerting-rules/templates/language.md:76]: `doc-validator` is not used in latest Grafana

[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/alerting-rules/templates/language.md:76]: `doc-validator` is not used in latest Grafana

File: docs/sources/alerting/configure-notifications/template-notifications/language.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+However, it's important to note that because notifications and alert rules operate in distinct context, some additional variables and functions are only available for either notification or alert rule templates. Refer to:
+
+- [Annotation and label template reference](ref:alert-rule-template-reference)
+- [Notification template reference](ref:notification-template-reference)
+
+This documentation provides an overview of the functions and operators of the Go template language that are available for both notification and alert rule templates.
+
+## Print
+
+To print the value of something, use `{{` and `}}`. You can print the value of a [variable](#variables), a field of a variable, the result of a function, or the value of dot.
```
[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/configure-notifications/template-notifications/language.md:58]: `doc-validator` is not used in latest Grafana

File: docs/sources/alerting/configure-notifications/template-notifications/language.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+At the start of notification templates, dot (`.`) refers to [Notification Data](ref:reference-notificationdata).
+
+```
+{{ .Alerts }}
+```
+
+In annotation and label templates, dot (`.`) is initialized with all alert data. It’s recommended to use the [`$labels` and `$values` variables](ref:alert-rule-template-reference-variables) instead to directly access the alert labels and query values.
+
+{{% admonition type="note" %}}
+Dot (`.`) might refer to something else when used in a [range](#range), a [with](#with), or when writing [templates](#templates) used in other templates.
```
[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/configure-notifications/template-notifications/language.md:80]: `doc-validator` is not used in latest Grafana

[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/configure-notifications/template-notifications/language.md:80]: `doc-validator` is not used in latest Grafana

[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/alerting/configure-notifications/template-notifications/language.md:80]: `doc-validator` is not used in latest Grafana

**Pull request merged**